### PR TITLE
BUG: Fixes for ARM architecture

### PR DESCRIPTION
**Pull request by larsoner**
Bad things found so far on my M1 machine:

- We can't rely on `LinAlgError` to trigger on `np.linalg.solve` for rank deficiency on arm64. I changed the `mxne` code to use a SVD, which actually seems faster (!?) so I think this is a tolerable change at least. This probably deserves an upstream bug report at some point, but isolating it isn't the easiest thing...
- We can't rely on SVD: https://github.com/numpy/numpy/issues/21756

The second one is a show stopper. I don't think we can safely release or advocate for M1-based installers until it's fixed. 

There are a bunch of `test_linalg.py` errors that I see locally that are likely related, but I don't want to bother looking into them until we can trust `svd`.

Helps with #10759

### CODE REVIEW

**Comment by larsoner** [comment]
@hoechenberger or @agramfort if either of you can replicate https://github.com/numpy/numpy/issues/21756 locally and can comment on the NumPy issue, it might help move things forward -- and also make sure I'm not doing something wrong!

**Comment by hoechenberger** [comment]
> * We can't rely on SVD: [BUG: linalg.svd broken on M1 / arm64 numpy/numpy#21756](https://github.com/numpy/numpy/issues/21756)

OMG this is horrible 😱 😱

**Comment by hoechenberger** [comment]
For all the others following this thread, the problem seems to be related to OpenBLAS, and switching to a different BLAS implementation seems to fix at least this here test case.

**Comment by larsoner** [comment]
Sorry I was fighting with https://gitlab.kitware.com/vtk/vtk/-/issues/18559 for the last hour instead :)

I think the most likely scenario is that it's an OpenBLAS issue that probably needs to be fixed at the OpenBLAS end. But now that it has been replicated on two machines, the NumPy folks should know how to move things forward. So I think we wait a couple of days and see if they make progress. In the meantime I think we wait on the M1 installers :(

**Comment by hoechenberger** [comment]
Can you try to conditionally use accelerate on ARM64 only, which is Apples own implementation of BLAS?

**Comment by larsoner** [comment]
> Can you try to conditionally use accelerate on ARM64 only, which is Apple's own implementation of BLAS?

I guess we could try this, but IIRC Accelerate was not really well supported by SciPy last time I saw it discussed. Feel free to try locally and run `make pytest` to see what fails. I'll give it a shot tomorrow.

**Comment by larsoner** [comment]
@agramfort feel free to merge if you're okay with the mxne changes

File: mne/inverse_sparse/mxne_optim.py [hunk 1]
```diff
@@ -220,34 +220,31 @@ def _mixed_norm_solver_bcd(M, G, alpha, lipschitz_constant, maxit=200,
...
+                # reliably raise LinAlgError here, so use SVD instead
+                u, s, _ = np.linalg.svd(C, hermitian=True)
+                if s[-1] <= 1e-6 * s[0]:
                     logger.debug("Iteration %d: LinAlg Error" % (i + 1))
-                else:
-                    c = z / z.sum()
-                    X_acc = np.sum(
-                        last_K_X[:-1] * c[:, None, None], axis=0
+                    continue
+                z = ((u * 1 / s) @ u.T).sum(0)
```
Comment (agramfort) [inline mne/inverse_sparse/mxne_optim.py:229]: @larsoner can you put a comment that this solves `z = np.linalg.solve(C, one_vec)`
it took me a few seconds to understand why

**Review by agramfort — COMMENTED** [review]
just a nitpick. than +1 for MRG on my side

thx @larsoner

**Comment by hoechenberger** [comment]
Great work, @larsoner!

**Pull request merged**